pipeliner/read_test.go

```diff
@@ -0,0 +1,233 @@
+package pipeliner
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/interpreter"
+	"github.com/amp-labs/connectors/common/jsonquery"
+	"github.com/amp-labs/connectors/test/utils/mockutils"
+	"github.com/amp-labs/connectors/test/utils/testutils"
+	"github.com/go-test/deep"
+)
+
+func TestRead(t *testing.T) { //nolint:funlen,gocognit,cyclop,maintidx
```

Comment:
It'd be nice to have all tests in one place, right now there's some test related files in the actual package, and some in the test/pipeliner directory.

Reply:
This is the original trend that I haven't changed in any way.
Any tests that make real calls to third party APIs live under `test/<connector-name>/<method-of-interest>/main.go`.

What I had introduced is tests of different level, type.

These tests work with prefetched data located in the form of `<connector-name>/test/<sample-server-ouptut>.json`.
Read/Write/ListObjectMetadata have "unit" tests side by side in a go methodology fashion.

Current test above uses mocked server response. It runs every time (CI) -- adding layer of protection unlike those under `test` directory, conversely the later can serve as an `example` on how to use a connector. Each with advantages/disadvantages.